cmd/winch/main.go

```diff
@@ -154,28 +168,60 @@ func main() {
 		}
 		g.Add(func() error {
 			log.Infof("listening for HTTP Plain on: %v", httpPlainListener.Addr().String())
-			err := httpWinchServer.Serve(httpPlainListener)
+			err := srv.Serve(httpPlainListener)
 			if err != nil {
 				return errors.Wrap(err, "http_plain server error")
 			}
 			return nil
 		}, func(error) {
 			log.Infof("\nReceived an interrupt, stopping services...\n")
 			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
-			err := httpWinchServer.Shutdown(ctx)
-			if err != nil {
+			if err := srv.Shutdown(ctx); err != nil {
 				log.WithError(err).Errorf("Failed to gracefully shutdown server.")
 			}
 			cancel()
 			httpPlainListener.Close()
 		})
 	}
+	// Setup optional external monitoring address.
```

Comment:
can you actually pull this into a separate func, this one is getting a bit long

Reply:
I would vote for leaving it that explicit - main is always long